src/gpl/src/nesterovBase.h

```diff
@@ -737,13 +707,13 @@ class BinGrid
   std::pair<int, int> getMinMaxIdxX(const Instance* inst) const;
   std::pair<int, int> getMinMaxIdxY(const Instance* inst) const;
 
-  const std::vector<Bin*>& bins() const;
+  std::vector<Bin>& bins();
+  const std::vector<Bin>& binsConst() const { return bins_; };
```

Comment:
Why is the non-const version now needed?  (Similarly where this API propagates upwards)

Reply:
bins is called in the non-const method `void NesterovBase::updateDensityForceBin()`

Reply:
I see the vector was constant but the pointers were not.  Now the vector itself must be mutable.